src/clients/entitlements/storage/in-memory/mappers/sources.mapper.ts

```diff
@@ -124,4 +126,17 @@ export class SourcesMapper {
 
     return NO_EXPIRE;
   }
+  private buildPlansSources(bundles: FeatureBundleTuple[], features: FeatureTuple[]): PlansSource {
+    const source: PlansSource = new Map();
+    features.forEach(([featureId, featureKey]) => {
+      bundles.forEach((tuple) => {
+        const featureIds = tuple[1];
+        if (featureIds.includes(featureId)) {
+          ensureArrayInMap(source, featureKey).push(mapPlanFromTuple(tuple));
+        }
+      });
+    });
```

Comment:
[suggestion]
it's not a big deal, but you could reduce the `O(N x M)` execution to `O(N + M)`, by going over features first, creating empty mappings (`featureKey` -> empty array), then going over bundles once and push the mapped bundle only for feature keys that we created empty array in previous step

Reply:
applied